cli/utils.ts

```diff
@@ -20,30 +22,29 @@ export type AnalyzedFunction = {
     returns: ValidatorJSON | null
 };
 
-export function getFunctionSpec(prod?: boolean, filePath?: string) {
-    if (filePath && prod) {
+export function getFunctionSpec(prod?: boolean, filePathOpt?: string) {
+    if (filePathOpt && prod) {
         console.error(`To use the prod flag, you can't provide a file path`);
         process.exit(1)
     }
     let content: string;
-    if (filePath && !fs.existsSync(filePath)) {
-        console.error(chalk.red(`File ${filePath} not found`));
+    if (filePathOpt && !fs.existsSync(filePathOpt)) {
+        console.error(chalk.red(`File ${filePathOpt} not found`));
         process.exit(1);
     }
-    if (filePath) {
-        content = fs.readFileSync(filePath, 'utf-8');
-    } else {
-        const flags = prod ? '--prod' : '';
-        let output: string;
+    let filePath = filePathOpt ?? path.join(os.tmpdir(), 'function-spec.json');
+    if (!filePathOpt) {
         try {
-            output = execSync(`npx convex function-spec ${flags}`).toString();
+            const flags = prod ? "--prod" : "";
+            execSync(`npx convex function-spec ${flags} > ${filePath}`).toString();
```

Comment:
I think the `>` works in all the Windows shells? but it would be nice not to rely on it, instead using a flag to the `convex function-spec` command to write to a file. If `filePath` has spaces in it the code here might not work.

Reply:
i agree. as currently configured, there is a `--file` flag that prints out to a file. but at the end we append the timestamp, so we would have to regex the filename or something. or i can update the convex CLI code to allow you to specify a file.

not sure what your thoughts are here

Reply:
Ah ok the convex CLI fix doesn't have to block this fix, but not working with spaces makes this worse that it was. Especially if full paths are involved lots of folks on Windows use spaces. Instead of execSync you can use a different child_process function that allows you to direct output to a file without any shell parsing going on.

Convex CLI fix sounds useful though, it's not ideal to not be able to specify the output filename. I think I'd rather have a flag for if you need the specialized behavior of appending a timestamp? Folks using this in CI flows will have to do the same glob trick if we don't allow specifying a specific filename.

Reply:
this seems reasonable. i am looking into the fix right now

Reply:
i just increased the buffer size of `execSync`. it looks like the default is 1MB, so I increased it to 10MB. This should mitigate the issues that we are seeing because we will be able to capture the whole output.

Reply:
actually, i found a way to write directly to a file instead!